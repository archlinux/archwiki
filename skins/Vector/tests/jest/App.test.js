const Vue = require( 'vue' );
const VueTestUtils = require( '@vue/test-utils' );
const App = require( '../../resources/skins.vector.search/App.vue' );

// @ts-ignore
Vue.directive( 'i18n-html', () => {} );

const defaultProps = {
	id: 'searchform',
	searchAccessKey: 'f',
	searchTitle: 'search',
	searchPlaceholder: 'Search MediaWiki',
	searchQuery: ''
};

const mount = ( /** @type {Object} */ customProps ) => {
	// @ts-ignore
	return VueTestUtils.shallowMount( App, {
		propsData: Object.assign( {}, defaultProps, customProps ),
		mocks: {
			$i18n: ( /** @type {string} */ str ) => ( {
				text: () => str
			} )
		}
	} );
};

describe( 'App', () => {
	it( 'renders a typeahead search component', () => {
		const wrapper = mount();
		expect(
			wrapper.element
		).toMatchSnapshot();
	} );
} );
