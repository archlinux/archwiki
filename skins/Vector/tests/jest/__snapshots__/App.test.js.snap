// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a typeahead search component 1`] = `
<wvui-typeahead-search-stub
  accesskey="f"
  aria-label="Search MediaWiki"
  buttonlabel="searchbutton"
  client="[object Object]"
  domain="localhost"
  formaction=""
  highlightquery="true"
  id="searchform"
  initialinputvalue=""
  placeholder="Search MediaWiki"
  searchpagetitle="Special:Search"
  showdescription="true"
  showthumbnail="true"
  suggestionslabel="searchresults"
  title="search"
  urlgenerator="[object Object]"
>
  <input
    name="title"
    type="hidden"
    value="Special:Search"
  />
   
  <input
    name="wprov"
    type="hidden"
    value="acrw1"
  />
  <span />
</wvui-typeahead-search-stub>
`;
